### PR TITLE
[DeadCode] Remove @return void on return self on RemoveUselessReturnTagRector

### PR DESCRIPTION
**Pull request by samsonasik**
Given the following code:

```php
    /**
     * @return void
     */
    public function resolve(): self
    {
        return $this;
    }
```

`@return void` should be removed. /cc @staabm

### CODE REVIEW

**Comment by staabm** [comment]
I think any phpdoc return-type which is not a subtype of the native type could be deleted

**Comment by samsonasik** [comment]
it is a bit complicated since it check generic type, also sub type like `positive-int` against native `int`, I will check more.

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
It seems cause error in other rule:

```diff
 There were 4 failures:

1) Rector\Tests\Php80\Rector\FunctionLike\MixedTypeRector\MixedTypeRectorTest::test with data set #5
Failed on fixture file "skip_mixed_type_with_nullable_scalar_param.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 
 final class SkipMixedTypeWithNullableScalarParam
 {
-    /**
-     * @param ?scalar $value
-     */
     public function __construct(
```

I will check more...

**Comment by samsonasik** [comment]
@staabm this is more complex than I though, closing for now.

File: rules/DeadCode/PhpDoc/DeadReturnTagValueNodeAnalyzer.php [hunk 1]
```diff
@@ -46,15 +49,15 @@ public function isDead(ReturnTagValueNode $returnTagValueNode, ClassMethod $clas
             $returnTagValueNode->type,
             $classMethod,
         )) {
-            return false;
+            return $returnTagValueNode->type instanceof IdentifierTypeNode && (string) $returnTagValueNode->type === 'void';
```
Comment (samsonasik) [inline rules/DeadCode/PhpDoc/DeadReturnTagValueNodeAnalyzer.php:52]: @staabm this is the solution I can found right now, due to sub type and generic usage, union complexity :), `void` mostly a different part so should be ok for this case ;)

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
@TomasVotruba let's merge it to have faster feedback to test ;)

**Pull request merged**